src/twisted/internet/test/test_tcp.py

```diff
@@ -963,6 +973,35 @@ def count():
         """
 
 
+@attr.s(auto_attribs=True)
+class SourceCacheForCoverage:
```

Comment:
Thanks for the fix.

I have checked the commit message but I am not sure why we need this.

I think that we need to add some info as part of the docstring of this class.

Is this a defect in `coverage` ... or is there any way in which coverage can make this easier ? Maybe we can link to an upstream GitHub issue.

```suggestion
class SourceCacheForCoverage:
    """"
    This is used to allow coverage to keep opening files even after EMFILE as part of _ExhaustsFileDescriptors.
    ADD MORE INFO ABOUT WHAT IS GOING ON HERE>
    """"
```

Reply:
> I have checked the commit message but I am not sure why we need this.
> 
> I think that we need to add some info as part of the docstring of this class.

Yeah, this is extremely esoteric and you're correct that I should report it upstream, link to it, and describe in detail why it is needed :)